lib/galaxy/config/__init__.py

```diff
@@ -133,37 +133,36 @@ def _set_config_directories():
                 self.config_dir = os.path.abspath(self.config_dir)
 
             self.data_dir = config_kwargs.get('data_dir')
-            # mutable_config_dir is intentionally not configurable. You can
-            # override individual mutable configs with config options, but they
-            # should be considered Galaxy-controlled data files and will by default
-            # just live in the data dir
             self.sample_config_dir = os.path.join(os.path.dirname(__file__), 'sample')
+            self.managed_config_dir = config_kwargs.get('managed_config_dir')
```

Comment:
If those options are NOT set by the admin (`managed_config_dir`, `data_dir`, etc.), they are eventually resolved w.r.t. an absolute path. If they ARE set by an admin, I think I would leave it up to the admin. There are many config options that are paths, and if some are made absolute by the system, while others are not, that may be unexpected behavior to an admin, I think? (all paths are equal, but some are more equal than the rest..)

Reply:
I think you are making an argument for making all paths absolute...

Reply:
It's ok if the admin sets it relative but is there harm in making it absolute internally?

Reply:
OK, no quick answer. I'm thinking of (a) ways it may break something and (b) at what point it would be safe to do so. For example, we want to call `strip_deprecated_dir` BEFORE we make user-set path absolute. There may be other issues like this one. Thinking...

Reply:
Most user-specified paths are made absolute when we call `_in_data_dir()` or sister methods on them, given that `self.data_dir` &c. are already absolute.

Reply:
Exactly. If the user sets `foopath='foo'`, and then it's passed to `_in_data_dir()`, you'll get something like `[abs data_dir path]/foo`. But if we *first* make that an abs path, then `_in_data_dir()` won't affect it and it'll remain `[abs foo path]`. So there will be no way for a user to set the a dir to 'foo' and have galaxy place it inside its `data_dir`. In other words, I'm not sure we are not trying to assume too much about user-set paths here.

Reply:
I'm not advocating for any other changes apart for this one, where the value comes directly from the admin.

Reply:
@ic4f does `managed_config_dir` get passed through `_in_data_dir()` later? If so then I suppose it renders @nsoranzo's suggestion moot, unless I misunderstand?

Reply:
it doesn't - I'm currently testing different config setups. So far it appears that if the admin sets it to some value that is a relative str, it stays relative.

Reply:
Double checking, but I think we, indeed, may want to make it absolute - most likely along with `data_dir` and `config_dir`. I'm still checking this..

Reply:
OK, so I'm making the change based on this premise: *if a config option is set by the user to a RELATIVE path, it must be ALWAYS resolved w.r.t some absolute directory by galaxy*.
I'll make the change for `managed_config_dir` here and will address the rest in a separate PR - not to convolute this one. I'll also document this (along with other config-related dev guidelines) - as soon as the rest of pending stuff in config is mostly done.